colossalai/engine/ophooks/_memtracer_ophook.py

```diff
@@ -1,3 +1,4 @@
+from re import S
```

Comment:
Did you use pre-commit hooks? I couldn't find where `S` is used.

Reply:
`S` is used for my debugging, I will delete it

Reply:
Do turn on pre-commit as it will check unused import.

Reply:
OK, I have turned it on.